### PR TITLE
Update k8s deployment document.

### PR DESCRIPTION
**Pull request by anfredette**
A large portion of this content came from the bpfd-operator README.md file, which will need to get trimmed down after this change is merged.

Signed-off-by: Andre Fredette <afredette@redhat.com>

### CODE REVIEW

File: docs/admin/k8s-deployment.md [hunk 1]
```diff
@@ -1,172 +1,404 @@
...
-    kubectl get pods -n cert-manager
-    NAME                                         READY   STATUS    RESTARTS   AGE
-    cert-manager-5dd59d9d9b-bbqbx                1/1     Running   0          22h
-    cert-manager-cainjector-8696fc9f89-rxtm7     1/1     Running   0          22h
-    cert-manager-webhook-7d4b5b8c56-gt6xb        1/1     Running   0          22h
-    ```
+The Bpfd-Operator can be used to deploy and manage bpfd within a kubernetes cluster.
 
-## Bpfd Component install
+![bpfd on K8s](./bpfd-on-k8s.jpg)
```
Comment (Billy99) [inline docs/admin/k8s-deployment.md:13]: Drawing needs to be updated to remove `E` from EbpfProgramConfig and EbpfProgram.

File: docs/admin/k8s-deployment.md [hunk 2]
```diff
@@ -1,172 +1,404 @@
...
+BPFD_IMG=<your/image/url> BPFD_AGENT_IMG=<your/image/url> BPFD_OPERATOR_IMG=<your/image/url> make run-on-kind
 ```
 
-4. Install bpfd `ServiceAccount`, `ClusterRole`, and `ClusterRoleBinding`.
+To push and test out any local changes simply run:
 
 ```bash
-kubectl apply -f ./packaging/kubernetes-deployment/bpfd-core/bpfd-agent-rbac.yaml
+make kind-reload-images
 ```
```
Comment (Billy99) [inline docs/admin/k8s-deployment.md:80]: Add something like:

To tear down the kind cluster, run:

```bash
kind delete cluster -n bpfd-deployment
```

The default name of the created cluster is `bpfd-deployment`.
To retrieve all the created kind clusters, use `kind get clusters`.

File: docs/admin/k8s-deployment.md [hunk 3]
```diff
@@ -1,172 +1,404 @@
...
+
+To delete the CRDs from the cluster:
+
+```sh
+make uninstall
+```
+
+## Undeploy controller
+
+UnDeploy the controller to the cluster:
```
Comment (Billy99) [inline docs/admin/k8s-deployment.md:392]:
```suggestion
Undeploy the controller from a vanilla K8s Deployment cluster:
```
There is also a `make deploy` command. Not sure if you want to document it like the reinstall of the CRD or not.

**Review by Billy99 — COMMENTED** [review]
Looks good. A couple minor comments

File: docs/admin/k8s-deployment.md [hunk 4]
```diff
@@ -1,172 +1,404 @@
...
+```
+
+3. Deploy the operator and agent to a cluster with the image specified by
+   `BPFD_AGENT_IMG` and `BPFD_OPERATOR_IMG`:
+
+```sh
+make deploy BPFD_OPERATOR_IMG=<some-registry>/bpfd-operator:tag BPFD_AGENT_IMAGE=<some-registry>/bpfd-agent:tag
+```
+
+## Uninstall CRDs
```
[the remark this answers is not in the document]
Comment (astoycos) [reply docs/admin/k8s-deployment.md:382]: No idea :) I assume the programconfig will be automatically deleted

File: docs/admin/k8s-deployment.md [hunk 5]
```diff
@@ -1,172 +1,417 @@
-# How to Manually Deploy bpfd on Kubernetes
+# How to Deploy bpfd on Kubernetes
 
-## Pre-requsites
+One of the key use cases for bpfd is to support the deployment of BPF programs
+in kubernetes clusters.  The simplest way to do this is to use the
```
Comment (Billy99) [inline docs/admin/k8s-deployment.md:4]: Global comment, kubernetes -> Kubernetes

File: docs/admin/k8s-deployment.md [hunk 6]
```diff
@@ -1,172 +1,417 @@
...
 ```
 
-5. Install bpfd daemonset which contains the `bpfd` and `bpfd-agent` processes.
-   This runs a `bpfd-xxxxx` pod on each node, where each pod contains a `bpfd` and `bpfd-agent` container.
+To tear down the kind cluster, run:
+```bash
+kind delete cluster -n bpfd-deployment
+```
 
+The default name of the created cluster is bpfd-deployment.
```
Comment (Billy99) [inline docs/admin/k8s-deployment.md:86]:
```suggestion
The default name of the created cluster is `bpfd-deployment`.
```

**Review by Billy99 — APPROVED** [review]
/lgtm
Super picky nits.

**Comment by anfredette** [comment]
@Billy99, I think I've addressed all of your comments.  Thanks for the thorough review!

**Review by Billy99 — APPROVED** [review]

**Pull request merged**